containers/unit_tests/TestWithoutInitializing.hpp

```diff
@@ -113,43 +113,60 @@ TEST(TEST_CATEGORY, resize_realloc_no_alloc_dualview) {
 }
 
 TEST(TEST_CATEGORY, resize_exec_space_dualview) {
-  using namespace Kokkos::Test::Tools;
-  listen_tool_events(Config::DisableAll(), Config::EnableFences(),
-                     Config::EnableKernels());
-  Kokkos::DualView<int*** * [1][2][3][4], TEST_EXECSPACE> bla("bla", 8, 7, 6,
-                                                              5);
-
-  auto success = validate_absence(
-      [&]() {
-        Kokkos::resize(
-            Kokkos::view_alloc(TEST_EXECSPACE{}, Kokkos::WithoutInitializing),
-            bla, 5, 6, 7, 8);
-        EXPECT_EQ(bla.template view<TEST_EXECSPACE>().label(), "bla");
-      },
-      [&](BeginFenceEvent event) {
-        if (event.descriptor().find("Kokkos::resize(View)") !=
-            std::string::npos)
-          return MatchDiagnostic{true, {"Found begin event"}};
-        return MatchDiagnostic{false};
-      },
-      [&](EndFenceEvent event) {
-        if (event.descriptor().find("Kokkos::resize(View)") !=
-            std::string::npos)
-          return MatchDiagnostic{true, {"Found end event"}};
-        return MatchDiagnostic{false};
-      },
-      [&](BeginParallelForEvent event) {
-        if (event.descriptor().find("initialization") != std::string::npos)
-          return MatchDiagnostic{true, {"Found begin event"}};
-        return MatchDiagnostic{false};
-      },
-      [&](EndParallelForEvent event) {
-        if (event.descriptor().find("initialization") != std::string::npos)
-          return MatchDiagnostic{true, {"Found end event"}};
-        return MatchDiagnostic{false};
-      });
-  ASSERT_TRUE(success);
-  listen_tool_events(Config::DisableAll());
+  // If we provide an execution space instance to Kokkos::resize with a DualView
+  // argument, it is used for both Views. Thus, the test only makes sense if the
+  // DefaultHostExecutionSpace can access TEST_EXECSPACE's memory space.
+  constexpr bool host_exec_space_can_access_memory_space =
+      Kokkos::SpaceAccessibility<Kokkos::DefaultHostExecutionSpace,
+                                 TEST_EXECSPACE::memory_space>::accessible;
+  if constexpr (!host_exec_space_can_access_memory_space) {
+    GTEST_SKIP()
+        << "skipping since host execution space can't access memory space";
+  } else {
+    using namespace Kokkos::Test::Tools;
+    listen_tool_events(Config::DisableAll(), Config::EnableFences(),
+                       Config::EnableKernels());
+    Kokkos::DualView<int*** * [1][2][3][4], TEST_EXECSPACE> bla("bla", 8, 7, 6,
+                                                                5);
+    auto success = validate_absence(
+        [&]() {
+          constexpr bool host_exec_space_can_access_memory_space =
+              Kokkos::SpaceAccessibility<
+                  Kokkos::DefaultHostExecutionSpace,
+                  TEST_EXECSPACE::memory_space>::accessible;
+          if constexpr (host_exec_space_can_access_memory_space) {
```

Comment:
I that not guarded by the `if constexpr` L122?

Reply:
No, I couldn't capture the `constexpr` variable as such and needed to repeat it.

Reply:
The compiler is too smart and sees the `static_assert(<EVAL-TO-FALSE>)` in the else-branch of the `if constexpr`.
If we template the unit test on `ExecutionSpace` it will work

Reply:
This gives
```
/var/jenkins/workspace/Kokkos/containers/unit_tests/TestWithoutInitializing.hpp: In lambda function:
/var/jenkins/workspace/Kokkos/containers/unit_tests/TestWithoutInitializing.hpp:133:16: error: declaration of 'host_exec_space_can_access_memory_space' shadows a previous local [-Werror=shadow]
  133 |           constexpr bool host_exec_space_can_access_memory_space =
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/jenkins/workspace/Kokkos/containers/unit_tests/TestWithoutInitializing.hpp:119:16: note: shadowed declaration is here
  119 |   constexpr bool host_exec_space_can_access_memory_space =
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```
Just template the unit test like I suggested, it works.